### ISSUE TITLE
Castle.Windsor is extremelly slow at resolving dependencies and also really hard？

### ISSUE THREAD

**Issue by feigaogit** [castleproject/Windsor#590]
#https://github.com/aspnetboilerplate/aspnetboilerplate/issues/5604
Is there a solution to this problem？
![image](https://user-images.githubusercontent.com/18181270/106120089-e6c83f00-6190-11eb-8db0-b5837e90d4f6.png)

**Comment by jonorossi** [comment]
@feigaogit news to me. An application where 75% of the time is spent in Windsor is definitely not normal.

A quick look over that issue talks about doing benchmarks with dotTrace or VS's tracing tools, which both significantly slow things down when any code generation of dynamic assemblies is involved (i.e. interception with DynamicProxy) because new assemblies are loaded into/modified in the runtime and tracing tools hook into runtime events. This is all via a very large ASP.NET Core project, a class with nearly 40 service dependencies, IIS, ASP.NET Boilerplate, EntityFramework, sounds like a heap of interceptors, the third party `Castle.Windsor.MsDependencyInjection` library, and all performing timing measured using Postman of a HTTP request through whatever other business logic or database the request uses.

I'm not trying to sound dismissive, but you've logged an issue here with no description and nothing concrete about Windsor for me to investigate. If you've got some code using just Windsor that is a performance problem, I can definitely look at it, but I can't spend my time attempting to pull apart a big web application. I also usually caution people that we are constrained by the performance of the runtime with regards to DynamicProxy and dynamically emitting code, and that often people use interceptors without an understanding of what Windsor is having to do under the hood, often times there are better extension points in web frameworks/pipelines or that should be handled by applications.

tl;dr please log a proper issue if you've found a performance problem.
